src/node_state.rs

```diff
@@ -171,9 +180,25 @@ impl NodeState {
         let blobs: Vec<UnsettledBlobMetadata> = tx
             .blobs
             .iter()
-            .map(|blob| UnsettledBlobMetadata {
-                blob: blob.clone(),
-                possible_proofs: vec![],
+            .enumerate()
+            .map(|(index, blob)| {
+                if let Ok(verifier) = (&blob.contract_name).try_into() {
```

Comment:
This reads a bit unclear to me to be honest, I think I'd like be more explicit about the type, maybe using TryInto::<>::try_into()